### ISSUE TITLE
including ubuntu recipe overwrites AWS mirrors

### ISSUE THREAD

**Issue by helgi** [bbg-cookbooks/monit-ng#46]
When using `cloud-init` or other similar tools they often change the base `sources.list` to point to a IaaS specific mirror (region specific or otherwise) for performance

```
-# See http://help.ubuntu.com/community/UpgradeNotes for how to upgrade to
-# newer versions of the distribution.
-deb http://us-west-2.ec2.archive.ubuntu.com/ubuntu/ trusty main
-deb-src http://us-west-2.ec2.archive.ubuntu.com/ubuntu/ trusty main
+deb http://us.archive.ubuntu.com/ubuntu trusty main restricted universe multiverse
+deb-src http://us.archive.ubuntu.com/ubuntu trusty main restricted universe multiverse
```

Is anyone aware of how to make it not totally overwrite the hostname there? Haven't looked too deeply into the ubuntu recipe this cookbook sources

**Comment by nathwill** [comment]
well, there's an attribute for the url base in the ubuntu cookbook, so you could probably control what it gets set to that way.

 as to the reason it's included by the monit-ng cookbook, it's because the monit package is generally in the multiverse repository, but many ubuntu systems only have "main" enabled.

we could of course make it a soft dependency, note the issue in the readme and then skip including ubuntu as part of this cookbook, leaving it up to the admin to handle the need to enable a repo with the monit package available...

personally, i don't care too much, as we don't use Ubuntu, so i'm not sure what Ubuntu users would find preferable. definitely open to user feedback here :smile:

**Comment by helgi** [comment]
I'll dig into it more then and see if I can come up with a solution that works for all without needing a soft dep if at all possible :)

**Comment by nathwill** [comment]
@helgi i guess one thing we could do is add a boolean attribute to make the ubuntu cookbook inclusion optional. that way we'd default to enabling the multiverse repository, but users for whom this might cause a problem can opt out and solve it in their preferred manner.

**Comment by helgi** [comment]
Not a bad idea. Had been hoping not to have to convolute the attributes on this cookbooks for this :)

So far I have solved it by running `ubuntu` first myself with the `archive_url` changed and only main / universe included. Could drop that in the README if I get happy with it.

Been looking at giving `ubuntu` some of the smarts around Cloud mirrors like `cloud-init` does but given `ubuntu` and `apt` are in flux I am not sure where to add it yet.

**Comment by nathwill** [comment]
okey doke. let's noodle on it a bit more then. if you've got a workable solution for now, i'm in no rush to address this, so we can leave this issue open until we get additional feedback or come up something else.

**Comment by helgi** [comment]
Yeah, sounds good. Tempted to mostly wait for upstream. See opscode-cookbooks/apt#131

**Comment by nathwill** [comment]
oh great. thanks for the link. definitely one to keep an eye on.
